semantic_router/llms/base.py

```diff
@@ -20,81 +20,139 @@ def __call__(self, messages: List[Message]) -> Optional[str]:
         raise NotImplementedError("Subclasses must implement this method")
 
     def _is_valid_inputs(
-        self, inputs: dict[str, Any], function_schema: dict[str, Any]
+        self, inputs: list[dict[str, Any]], function_schemas: list[dict[str, Any]]
     ) -> bool:
-        """Validate the extracted inputs against the function schema"""
+        """Determine if the functions chosen by the LLM exist within the function_schemas, 
+        and if the input arguments are valid for those functions."""
         try:
-            # Extract parameter names and types from the signature string
```

Comment:
Should we be validating with the same logic as we did for a single function call? But we simply iterate through the list of function schemas?

Reply:
Or - we keep a method that validates a single function schema (this seems like logical separation to me), and then around that we have the additional required logic here for handling multiple schemas

Reply:
Cool sounds good to me. Instead of the additional `_extract_parameter_info` function, I'll create a separate `_validate_function` function (which closely matches the original function) and which is called in a loop within `_is_valid_inputs`. 

We need to keep the following additional bits as we can now call multiple functions at a time, and we need to make sure that they exist in our schemas.

```

                # Check if 'function_name' and 'arguments' keys exist in each input dictionary
                if "function_name" not in input_dict or "arguments" not in input_dict:
                    logger.error("Missing 'function_name' or 'arguments' in inputs")
                    return False

                function_name = input_dict["function_name"]
                arguments = input_dict["arguments"]

                # Find the matching function schema based on function_name
                matching_schema = next((schema for schema in function_schemas if schema["name"] == function_name), None)
                if not matching_schema:
                    logger.error(f"No matching function schema found for function name: {function_name}")
                    return False
                    
```

Reply:
thanks